src/EvalGraph.purs

```diff
@@ -196,6 +201,33 @@ eval_progCxt (ProgCxt { primitives, mods, datasets }) =
       v <- eval γ e empty
       pure $ γ <+> maplet x v
 
+evalCmt :: forall m. MonadWithGraphAlloc m => Env Vertex -> DocOpt Expr Vertex -> m (DocOpt Val Vertex)
+evalCmt _ None = pure None
+evalCmt γ (Doc tokens) = Doc <$> sequence (map evalToken tokens)
+   where
+   evalToken :: DocCommentElem Expr Vertex -> m (DocCommentElem Val Vertex)
+   evalToken (Token s) = pure $ Token s
+   evalToken (Unquote e) = Unquote <$> (eval γ e empty)
+
+new'
```

Comment:
If this helper is specific to `EvalGraph`, there is no need to parameterise on `constr` -- just use `Val` directly